tools/interop-node/feeder/feeder.go

```diff
@@ -189,34 +172,25 @@ func (feeder *Feeder) HandleAbstain(ctx context.Context) error {
 	return nil
 }
 
-func (feeder *Feeder) gatherBlockDataString(chainIds []string, timestamp uint64) ([]string, error) {
-	blockDataList := make([]oracletypes.BlockData, len(chainIds))
-	for idx, cid := range chainIds {
-		bd, err := feeder.getOldestBlock(cid, timestamp)
-		if err != nil {
-			return nil, err
-		}
-
-		blockDataList[idx] = bd
-	}
-
-	return blockDataListToBlockDataString(blockDataList), nil
-}
-
 func (feeder *Feeder) getOldestBlock(chainId string, timestamp uint64) (oracletypes.BlockData, error) {
 	sub, ok := feeder.subscribers[chainId]
 	if !ok {
 		return oracletypes.BlockData{}, fmt.Errorf("chain client not found for chain id: %s", chainId)
 	}
 
 	bd, err := sub.GetOldestBlock(timestamp)
+	if err != nil {
```

Comment:
This is unnecessary, bd is not nil and caller should check if err is not nil.

Reply:
I don't understand why this is unnecessary. Can you elaborate please?
- `GetOldestBlock` returns `(oracletypes.BlockData, error)`, so it can potentially return an error.
- If there's an error, `bd` might be invalid or incomplete, which could cause issues later.